doc/requirements/io_class

```diff
@@ -19,65 +19,336 @@ It shall be possible to set a maximum size of an IO class.
+id: max_configured
+---
+
+The product shall allow to configure up to 33 IO classes.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Attributes
+id: attributes
+---
+
+The IO class configuration shall include a following attributes for each IO class:
+
+1. Id
+2. Classification rule
+3. Eviction priority
+4. Occupancy limit
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Allowed ids
+id: id_allowed_values
+---
+
+The product shall accept the IO class ids in range 0-32 inclusive.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Allowed eviction priorities
+id: eviction_priorities_allowed_values
+---
+
+The product shall accept the eviction priorities in range 0-255 inclusive.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Empty eviction priority
+id: empty_priority
+---
+
+The product shall allow to omit eviction priority during specifying IO class
+configuration.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Pinning IO class
+id: pinning
+---
+
+The product shall allow to pin IO classes.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Allowed occupancy values
+id: occupancy_allowed_values
+---
+
+The product shall accept integer and fixed point values for a max size of an IO class
+in range from 0.00 to 1.00 and interpret them as a fraction of a cache size.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Rules testing order
+id: rules_testing_order
+---
+
+IO request shall be tested against classification rules in the order they were
+specified in config.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Effective IO Class
+id: effective
+---
+
+Effective IO class for given IO request shall be the latest one for which
+the IO request fulfill the condition.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Last processed rule
+id: last_processed
+---
+
+The product shall allow to mark classification rule as the last eligible to process.
```

Comment:
This is unclear what it means. It looks like when we mark rule as "last" then we stop processing rules unconditionally.

Reply:
Maybe something along these lines (not sure about the word "terminating"):
The product shall allow to mark a classification rule as a terminating one. Upon matching I/O to a terminating classification rule no further classification is performed and the resolved (resulting?) I/O class is the terminating one.

Reply:
I think that we could point more explicitly that the termination takes place only if the IO request passes the check for the IO class marked as terminating. Btw this is a bit confusing that the "terminating" property of the IO class indicated by one of the classification rules.